RockWeb/Blocks/Finance/TransactionDetail.ascx.cs

```diff
@@ -1646,14 +1666,25 @@ private void ShowAccountDialog( Guid guid )
                 apAccount.SetValue( txnDetail.AccountId );
                 tbAccountAmount.Text = txnDetail.Amount.ToString( "N2" );
                 tbAccountSummary.Text = txnDetail.Summary;
+
+                if ( txnDetail.Attributes == null )
+                {
+                    txnDetail.LoadAttributes();
+                }
             }
             else
             {
                 apAccount.SetValue( null );
                 tbAccountAmount.Text = string.Empty;
                 tbAccountSummary.Text = string.Empty;
+
+                txnDetail = new FinancialTransactionDetail();
+                txnDetail.LoadAttributes();
             }
 
+            phAccountAttributeEdits.Controls.Clear();
+            Helper.AddEditControls( txnDetail, phAccountAttributeEdits, true );

```

Comment:
FYI. Validation is not getting triggered for any "required" attributes because the validationGroup parameter is missing. It should probably get the validation group from the modal and should look like this:
`Helper.AddEditControls( txnDetail, phAccountAttributeEdits, true, mdAccount.ValidationGroup );`

I'll add that after I merge this request if you're unable to update it.